Data/Graph/Inductive/Query/Cycles.hs

```diff
@@ -0,0 +1,280 @@
+{- |
+   Module      : Data.Graph.Inductive.Query.Cycles
+   Description : Algorithms for finding all cycles.
+   Copyright   : (c) Gabriel Hjort Blindell 2017
+   License     : 2-Clause BSD
```

Comment:
FGL is licensed under 3-Clause BSD; in interests of using the same license for all, would you mind changing this?

(And some code from here is from Graphalyze, which isn't under your Copyright.)